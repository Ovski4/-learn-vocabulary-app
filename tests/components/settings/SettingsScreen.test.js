import React from 'react';
import { createStore } from 'redux';
import reducer from '../../../src/reducers/index';
import SettingsScreen from '../../../src/components/settings/settingsScreen/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('SettingsScreen component', () => {

    const getStore = (locale) => {
        return createStore(reducer, {
            config: {
                locale: locale
            }
        });
    };

    it('Should render the settings screen with the english locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('en')}>
                <SettingsScreen />
            </Provider>).toJSON();

        expect(rendered).toMatchSnapshot();
    });

    it('Should render the settings screen with the french locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('fr')}>
                <SettingsScreen />
            </Provider>).toJSON();

        expect(rendered).toMatchSnapshot();
    });

});
