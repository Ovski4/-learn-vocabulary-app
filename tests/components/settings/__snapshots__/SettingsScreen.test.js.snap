// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen component Should render the settings screen without crashing with english locale 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Settings
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#929292",
          "fontSize": 16,
        }
      }
    >
      Language
      :
    </Text>
    <View
      style={
        Object {
          "height": 40,
          "paddingLeft": 20,
          "width": 200,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "English",
              "textColor": undefined,
              "value": "en",
            },
            Object {
              "label": "Français",
              "textColor": undefined,
              "value": "fr",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 5,
        },
        Object {
          "bottom": 0,
          "paddingBottom": 30,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#dcdcdc",
          "textAlign": "center",
        }
      }
    >
      v1.2.0
    </Text>
  </View>
</View>
`;

exports[`SettingsScreen component Should render the settings screen without crashing with french locale 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Paramètres
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#929292",
          "fontSize": 16,
        }
      }
    >
      Langue
      :
    </Text>
    <View
      style={
        Object {
          "height": 40,
          "paddingLeft": 20,
          "width": 200,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "English",
              "textColor": undefined,
              "value": "en",
            },
            Object {
              "label": "Français",
              "textColor": undefined,
              "value": "fr",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={1}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 5,
        },
        Object {
          "bottom": 0,
          "paddingBottom": 30,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#dcdcdc",
          "textAlign": "center",
        }
      }
    >
      v1.2.0
    </Text>
  </View>
</View>
`;
